src/asset_util/src/lib.rs

```diff
@@ -282,22 +335,26 @@ fn response_hash(headers: &[HeaderField], body_hash: &Hash) -> Hash {
     response_hash
 }
 
-pub fn collect_assets_recursive(
-    dir: &Dir,
-    html_transformer: fn(&str) -> String,
-) -> Vec<(String, Vec<u8>, ContentEncoding, ContentType)> {
+#[derive(Debug, Clone)]
+pub struct Asset {
```

Comment:
nit: maybe move closer to the top (near `CertifiedAssets` and `CertifiedAsset`), and comment, so that it is clear what are the functions that these structs play?